src/ViewHelpers/RenderViewHelper.php

```diff
@@ -98,17 +115,27 @@ public function render() {
 		$arguments = (array) $this->arguments['arguments'];
 		$optional = (boolean) $this->arguments['optional'];
 		$contentAs = $this->arguments['contentAs'];
+		$delegate = $this->arguments['delegate'];
 		$tagContent = $this->renderChildren();
 
 		if ($contentAs !== NULL) {
 			$arguments[$contentAs] = $tagContent;
 		}
 
 		$content = '';
-		if ($partial !== NULL) {
+		if ($delegate !== NULL) {
+			if (!is_a($delegate, ParsedTemplateInterface::class, TRUE)) {
+				throw new \InvalidArgumentException(sprintf('Cannot render %s - must implement ParsedTemplateInterface!', $delegate));
+			}
+			$renderingContext = clone $this->renderingContext;
+			$renderingContext->getVariableProvider()->setSource($arguments);
+			$content = (new $delegate())->render($renderingContext);
+		} elseif ($partial !== NULL) {
 			$content = $this->viewHelperVariableContainer->getView()->renderPartial($partial, $section, $arguments, $optional);
 		} elseif ($section !== NULL) {
 			$content = $this->viewHelperVariableContainer->getView()->renderSection($section, $arguments, $optional);
+		} elseif (!$optional) {
```

Comment:
We're trying to move away from that explicitness unless the function potentially returns something not-boolean - which `is_a` does not. The `!== NULL` stuff is a little different since it checks (uncritically) if a VH argument was not provided and still has the default value. Oh and for `$optional` specifically it should already be cast to boolean above ;)